app/assets/javascripts/tasks_index.coffee

```diff
@@ -8,18 +7,23 @@ ready = ->
   initializeInputFieldEnterCallback()
 
 $(document).on('turbolinks:load', ready)
+window.select2_locales_loaded.then(initializeSelect2)
+console.log("task_index select2_locales_loaded then registered")
 
 initializeDynamicHideShow = ->
   $('body').on 'click', '.more-btn-wrapper', (event) ->
     event.preventDefault()
     toggleHideShowMore $(this)
 
 initializeSelect2 = ->
```

Comment:
> Actually I cannot figure it out. Maybe you see the problem? I pushed my current state. The problem is that only the label select2 gets initialized and the others are not.

Here's the issue: You're first setting up the promise chain by defining that the function `initializeSelect2` should be called. However, at that time, the function `initializeSelect2` is not yet known, since it only gets defined thereafter.

Reply:
Right, I was not aware that this is a thing in JS :)